pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractDispatcherMultipleConsumers.java

```diff
@@ -119,17 +119,13 @@ public Consumer getNextConsumer() {
             return null;
         }
 
-        if (currentConsumerRoundRobinIndex >= consumerList.size()) {
-            currentConsumerRoundRobinIndex = 0;
-        }
-
         int currentRoundRobinConsumerPriority = consumerList.get(currentConsumerRoundRobinIndex).getPriorityLevel();
 
         // first find available-consumer on higher level unless currentIndex is not on highest level which is 0
         if (currentRoundRobinConsumerPriority != 0) {
             int higherPriorityConsumerIndex = getConsumerFromHigherPriority(currentRoundRobinConsumerPriority);
             if (higherPriorityConsumerIndex != -1) {
-                currentConsumerRoundRobinIndex = higherPriorityConsumerIndex + 1;
+                currentConsumerRoundRobinIndex = (higherPriorityConsumerIndex + 1) % consumerList.size();
```

Comment:
Should we need check  the consumer validation use `org.apache.pulsar.broker.service.AbstractDispatcherMultipleConsumers#isConsumerAvailable`？

Reply:
next loop, It will check